[Hackathon 5th No.73] ToT

ErnestinaQiu opened the pull request:
1. finish meta/llama2 version
2. fast start in readme.md

<!-- Demo: https://github.com/PaddlePaddle/PaddleNLP/pull/26 -->
### PR types
<!-- One of [ New features  ] -->
add tree-of-thought

### PR changes
<!-- One of [ Models | Docs |  ] -->

### Description
<!-- add tree-of-thought by using meta/llama2 -->

ErnestinaQiu committed:
Hackathon TASK73 ToT

1. finish meta/llama2 version

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PaddlePaddle/PaddleNLP?pullRequest=7660) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PaddlePaddle/PaddleNLP?pullRequest=7660) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: ErnestinaQiu<br/>:x: root<br/><hr/>**root** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PaddlePaddle/PaddleNLP?pullRequest=7660) it.</sub>

ErnestinaQiu committed:
update readme tutorial

ErnestinaQiu committed:
modify according to Lint

ErnestinaQiu committed:
modify according Link

1. resolve one unused variable

ErnestinaQiu committed:
Delete LICENSE

ErnestinaQiu committed:
Update LICENSE

ErnestinaQiu committed:
black format

ErnestinaQiu committed:
isort format

ErnestinaQiu committed:
Update search_crosswords-dfs.ipynb

ErnestinaQiu committed:
update files formats

ErnestinaQiu committed:
Update LICENSE

ErnestinaQiu committed:
Update LICENSE

ErnestinaQiu committed:
Update LICENSE

ErnestinaQiu committed:
Update LICENSE

ErnestinaQiu committed:
delete test data

w5688414 commented:
1. 删除关于torch 的llama的代码，该用paddlenlp的llama
2. 删除所有的数据集，只需要在README里面添加数据集的链接即可

w5688414 commented on pipelines/examples/agents/tree-of-thought-llm/llama2/llama/.gitignore:
```diff
@@ -0,0 +1,160 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
```
.gitignore删除

w5688414 commented on pipelines/examples/agents/tree-of-thought-llm/llama2/llama/CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,80 @@
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment
```
CONDUCT删除，这个与tot无关

w5688414 commented on pipelines/examples/agents/tree-of-thought-llm/llama2/llama/USE_POLICY.md:
llama目录的所有文件删除，只保留tot相关的代码

w5688414 commented on pipelines/examples/agents/tree-of-thought-llm/src/tot/data/crosswords/mini0505.json:
```diff
@@ -0,0 +1,6398 @@
+            "O",
+            "T",
+            "O",
+            "R",
```
数据集全部删除

ErnestinaQiu committed:
delete some unnecessary files

1. delete some unnecessary files according to comments.

ErnestinaQiu committed:
add paddlenlp-llama2

1. add llama2 in paddlenlp

ErnestinaQiu committed:
fix one bug

ErnestinaQiu committed:
fix outputs bug

1. format data structure

ErnestinaQiu committed:
delete meta/llama2

ErnestinaQiu commented:
![teaser](https://github.com/PaddlePaddle/PaddleNLP/assets/48557439/30f9e365-398a-4822-b3c2-a0768f70e310)

ErnestinaQiu committed:
modify according to comments

1. add acknow into readme
2.change png into url in readme
3. add all the models supported by paddlenlp

w5688414 commented on pipelines/examples/agents/tree-of-thought/README.md:
````diff
@@ -0,0 +1,101 @@
+2. Install `tot` package in two ways:
+- Option 1: Install from PyPI
+```bash
+pip install tree-of-thoughts-llm
````
readme把安装tree-of-thoughts相关的去除

w5688414 commented on pipelines/examples/agents/tree-of-thought/requirements.txt:
```diff
@@ -0,0 +1,51 @@
+networkx==3.1
+numpy==1.24.3
+nvidia-cublas-cu11==11.10.3.66
+nvidia-cuda-cupti-cu11==11.7.101
```
requirements要写这么多？nvidia的那些不用，去除torch等相关的依赖

w5688414 commented on pipelines/examples/agents/tree-of-thought/src/llama2/__init__.py:
```diff
@@ -0,0 +1,2 @@
+# coding=utf8, ErnestinaQiu
+from src.llama2.llama import llamaChatCompletion
```
目录从llama2改成llm

w5688414 commented on .gitignore:
```diff
@@ -123,4 +123,7 @@ FETCH_HEAD
-./ppdiffusers/ppdiffusers/version.py
\ No newline at end of file
+./ppdiffusers/ppdiffusers/version.py
+test_params.py
```
去除gitignore

w5688414 commented on pipelines/examples/agents/tree-of-thought/README.md:
```diff
@@ -0,0 +1,101 @@
+# Tree of Thoughts (ToT)
+
+![teaser](https://github.com/PaddlePaddle/PaddleNLP/pull/7660#issuecomment-1867279637)
```
插入github生成的图片的链接，而不是图片所在的issue的位置

```
![teaser](https://github.com/PaddlePaddle/PaddleNLP/assets/48557439/30f9e365-398a-4822-b3c2-a0768f70e310)
```

w5688414 commented:
pre-commit install 
检查一下codestyle

ErnestinaQiu committed:
change according to comments

ErnestinaQiu committed:
Delete .gitignore

ErnestinaQiu committed:
Create .gitignore

Commit:
Move directory

Commit:
Add tree of thoughts scripts

ErnestinaQiu committed:
add first dir

ErnestinaQiu committed:
Merge branch 'develop' of https://github.com/ErnestinaQiu/tot into develop

w5688414 commented on pipelines/examples/tree-of-thought/README.md:
```diff
@@ -0,0 +1,188 @@
+    --n_select_sample 1 \
+    --prompt_sample cot \
+    --temperature 1.0 \
+    ${@}
```
如果没啥实际意义，就删除

w5688414 reviewed (commented):
LGTM

w5688414 commented:
1.后续给出在paddlenlp的llama模型上的实验效果，给出参数指标，写进readme里面

ErnestinaQiu committed:
add note

ErnestinaQiu committed:
Update README.md

add test results of facebook/llama-2-7b-chat and llama-2-13b-chat

w5688414 commented on pipelines/examples/tree-of-thought/src/tot/tasks/crosswords.py:
```diff
@@ -0,0 +1,287 @@
+            res = res.split("\n")[-1].strip()
+            if res in count:
+                count[res] += 1
+        print(count)
```
把print打印去除

w5688414 commented on pipelines/examples/tree-of-thought/src/tot/tasks/game24.py:
```diff
@@ -0,0 +1,114 @@
+        current_numbers = get_current_numbers(y if y else x)
+        if current_numbers == "24":
+            prompt = cot_prompt.format(input=x) + "Steps:" + y
+            # print([prompt])
```
去除print

w5688414 commented on pipelines/examples/tree-of-thought/src/tot/tasks/text.py:
```diff
@@ -0,0 +1,121 @@
+                scores.append(score)
+            else:
+                print(f"------------------score no match: {[score_output]}")
+        print(scores)
```
代码里面不要出现print，可以使用logging输出

w5688414 commented on pipelines/examples/tree-of-thought/src/tot/tasks/game24.py:
```diff
@@ -0,0 +1,114 @@
+        if sorted(numbers) != sorted(problem_numbers):
+            return {"r": 0}
+        try:
+            # print(sympy.simplify(expression))
```
去除

w5688414 commented on pipelines/examples/tree-of-thought/README.md:
```diff
@@ -0,0 +1,193 @@
+git clone https://github.com/PaddlePaddle/PaddleNLP.git
+cd pipelines/examples/agents
+pip install -r requirements.txt
+pip install -e .  # install `tot` package
```
目录位置移动了，另外，去除`pip install -e .  # install `tot` package`

w5688414 commented on pipelines/examples/tree-of-thought/README.md:
```diff
@@ -0,0 +1,193 @@
+
+## 论文实验
+
+通过 ``sh scripts/{game24, text, crosswords}/{standard_sampling, cot_sampling, bfs}.sh`` 运行实验，除了在填字游戏中我们使用了 ToT 的 DFS 算法，可以通过 ``scripts/crosswords/search_crosswords-dfs.ipynb`` 运行。
```
脚本去除了，把README文档更新一下

w5688414 commented on pipelines/examples/tree-of-thought/requirements.txt:
```diff
@@ -0,0 +1,37 @@
+sympy==1.12
+termcolor==2.3.0
+tqdm==4.65.0
+triton==2.0.0
```
triton也需要？简化一下requirements.txt，把不需要和重复的依赖删除，另外把paddlenlp，paddle的依赖加入进去

ErnestinaQiu committed:
Update requirements.txt

delete unnecessary packages

ErnestinaQiu committed:
Update demo.py

add Ernie

ErnestinaQiu committed:
Update .gitignore

delete pyproject.toml

ErnestinaQiu committed:
Update run.py

add Ernie

ErnestinaQiu committed:
Update __init__.py

add Ernie

ErnestinaQiu committed:
chat templates

ErnestinaQiu committed:
add Ernie

ErnestinaQiu committed:
Update llama.py

兼容Ernie

ErnestinaQiu committed:
Update bfs.py

兼容Ernie

ErnestinaQiu committed:
Update models.py

兼容Ernie

ErnestinaQiu committed:
Update run.py

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

w5688414 reviewed (dismissed):
LGTM

w5688414 commented on pipelines/examples/tree-of-thought/README.md:
```diff
@@ -0,0 +1,206 @@
+* 在 ``tot/tasks/`` 中设置一个新的任务类和任务文件在 ``tot/data/`` 中。查看 ``tot/tasks/game24.py`` 以获取示例。将任务添加到 ``tot/tasks/__init__.py`` 中。
+* 在 ``tot/prompts/`` 中设置任务特定的提示。查看 ``tot/prompts/game24.py`` 以获取示例。根据任务的性质，选择 ``--method_generate`` (choices=[``sample``, ``propose``]) 和 ``--method_evaluate`` (choices=[``value``, ``vote``]) 及其相应的提示。
+
+## 测试结果
```
测试结果有2个，删去其中的一个

w5688414 commented on pipelines/examples/tree-of-thought/src/llm/ernie_bot.py:
```diff
@@ -0,0 +1,79 @@
+        return new_out
+
+
+# 检查字符串中是否存在中文 和 数字
```
注释放到函数里面

w5688414 commented on pipelines/examples/tree-of-thought/src/llm/ernie_bot.py:
```diff
@@ -0,0 +1,79 @@
+
+Ernie_llm_list = ["ernie-3.5", "ernie-4.0"]
+
+erniebot.api_type = "aistudio"
```
api_type,和access_token两个可以通过环境变量控制，不需要写到这里面，删除

ErnestinaQiu committed:
删掉重复的“测试结果”

ErnestinaQiu committed:
删除Ernie的token，设置环境变量解决

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
format style

ErnestinaQiu committed:
删除注释掉的代码

w5688414 reviewed (approved):
LGTM
